### PR TITLE
feat(visor): Update chart for visor@v0.4

### PR DESCRIPTION
**Pull request by placer14**
This PR does a few things on the Visor chart:
- removes `run` options in favor of `walk` and `watch` options
- moves non-sensitive options to be args instead of envs to ensure runtime failure on unexpected inputs (instead of silently failing when the env interface changes)
- adds namespace checks on nested values
- adds validations on required or mutually exclusive inputs
- add some organization and comments

### CODE REVIEW

File: charts/visor/templates/deployment.yaml [hunk 1]
```diff
@@ -65,28 +76,95 @@ spec:
...
+          - "--name={{ .Values.pgAppName }}"
+          {{- end }}
+
+          - "--allow-schema-migration={{ .Values.allowAutomaticMigration }}"
+
+          # lens config
+          {{- if .Values.lens }}
+          {{- if and .Values.lens.lotusAPI }}
+          - "--lens=lotus"
+          {{- $lotusAPIMultiaddr := "/dns/{{ .Release.Name }}-lotus-daemon.{{ .Release.Namespace }}.svc.cluster.local/tcp/1234" -}}
```
Comment (iand) [inline charts/visor/templates/deployment.yaml:116]: I don't think we'll ever use this automatic lotus address. I'd prefer to remove it and add a check to ensure that there is a multiaddr set in the values

File: charts/visor/templates/deployment.yaml [hunk 2]
```diff
@@ -65,28 +76,95 @@ spec:
...
+          {{- if .Values.lens.lotusAPI.lotusAPIMultiaddr -}}
+            {{- $lotusAPIMultiaddr := .Values.lens.lotusAPI.lotusAPIMultiaddr -}}
+          {{- end }}
+          - "--repo=$(LOTUS_API_TOKEN):{{ $lotusAPIMultiaddr }}"
```
Comment (iand) [inline charts/visor/templates/deployment.yaml:120]: Think this should be --api

Comment (iand) [reply charts/visor/templates/deployment.yaml:120]: Also, I'd like to split --api into two separate flags in visor: --api and --api-token. Makes the separation cleaner and means the token can very easily come from a secret without munging into a composite value.

**Review by iand — COMMENTED** [review]
LGTM (aside from --api comment). So much simpler

File: charts/visor/templates/deployment.yaml (hunk 2 above)
Comment (placer14) [reply charts/visor/templates/deployment.yaml:120]: Let's include this in the next release.

Comment (placer14) [reply charts/visor/templates/deployment.yaml:120]: Ended up walking this change back for now and will continue the existing env-var approach.

**Comment by ognots** [comment]
this new format is great! the re-arrangement of values are logical and easier to grok configuration. only left one nit comment about a stray comment in values.yaml self documentation

**Review by ognots — APPROVED** [review]

**Pull request merged**